x/oracle/keeper/grpc_query.go

```diff
@@ -155,3 +156,107 @@ func (k Querier) RequestSearch(c context.Context, req *types.QueryRequestSearchR
 func (k Querier) RequestPrice(c context.Context, req *types.QueryRequestPriceRequest) (*types.QueryRequestPriceResponse, error) {
 	return &types.QueryRequestPriceResponse{}, nil
 }
+
+func (k Querier) RequestVerification(c context.Context, req *types.QueryRequestVerificationRequest) (*types.QueryRequestVerificationResponse, error) {
+	// Request should not be empty
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	// Provided chain ID should match validator's chain ID
+	if ctx.ChainID() != req.ChainId {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("provided chain ID does not match the validator's chain ID; expected %s, got %s", ctx.ChainID(), req.ChainId))
```

Comment:
codes.BadRequest (400) on this case should be better.

does not match the validator's chain => does not match the current chain ID

Reply:
There is no Bad Request error for gRPC status code. However, there is a mapping here:

https://grpc.github.io/grpc/core/md_doc_http-grpc-status-mapping.html

Shall we use `codes.INTERNAL` instead as an mapping of `Bad Request`?

Reply:
I think this link help us more to use code https://pkg.go.dev/google.golang.org/grpc@v1.36.1/codes#Code

Reply:
But not internal for sure. FailedPrecondition?

Reply:
Sounds good. Will use FailedPrecondition.